### ISSUE TITLE
Components 'classic mvc' example is not correct.

### ISSUE THREAD

**Issue by rajaraodv** [lhorie/mithril.js#634]
http://lhorie.github.io/mithril/components.html#classic-mvc
When we save, we send reference to the contact in ContactForm to be added to the list. Since it's only a reference ( this.contact = m.prop(new Contact())) , every time we add new data, all those new items will have the same value! 

You can test it by adding two more contacts here:  http://jsfiddle.net/rajaraodv/ptp65n8q/31/

```
var ContactForm = {
    controller: function(args) {
        this.contact = m.prop(new Contact())
        this.save = function(contact) {
            Observable.trigger("saveContact", {contact: contact})
        }
    },
    view: function(ctrl, args) {
        var contact = ctrl.contact()

        return m("form", [
            m("label", "Name"),
            m("input", {oninput: m.withAttr("value", contact.name), value: contact.name()}),

            m("label", "Email"),
            m("input", {oninput: m.withAttr("value", contact.email), value: contact.email()}),

            m("button[type=button]", {onclick: ctrl.save.bind(this, contact)}, "Save")
        ])
    }
}
```

**Comment by rajaraodv** [comment]
Solution would be to copy form-contact's data (see copyContact) and then resetting form-contact's data http://jsfiddle.net/rajaraodv/ptp65n8q/32/

**Comment by pelonpelon** [comment]
I guess I was busy working on a solution when this issue was posted.
see PR #636

**Comment by tivac** [comment]
This appears to have been fixed in 4b2ec49ca9c66148b5c80511085672ab184249da
